src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -42,25 +55,30 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner = owner;
                 _subjectBuffer = subjectBuffer;
 
-                // Register to hear about diagnostics changing.  When we're notified about new
-                // diagnostics (and those diagnostics are for our buffer), we'll ensure that
-                // we have an underlying tagger responsible for asynchronously handling diagnostics
-                // from the owner of that diagnostic update.
-                _owner._diagnosticService.DiagnosticsUpdated += OnDiagnosticsUpdated;
+                var document = _subjectBuffer.AsTextContainer().GetOpenDocumentInCurrentContext();
```

Comment:
just FYI - there is also an event

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace_Events.cs,164

...

combine with workspace can also be changed with registered text buffer (WorkspaceRegistration), the way you did seems simpler though.